src/analysis/special_functions/integrals.lean

```diff
@@ -0,0 +1,103 @@
+
+This file contains proofs of the integrals of various simple functions, including `pow`, `exp`,
+`inv`/`one_div`, `sin`, `cos`, and `λ x, 1 / (1 + x^2)`.
+
+With these lemmas, many simple integrals can be computed by `simp` or `norm_num`. Scroll to the
+bottom of the file for examples.
+
+This file is incomplete; we are working on expanding it.
+-/
+
+open real set interval_integral
+variables {a b : ℝ}
+
+@[simp]
+lemma integral_pow (n : ℕ) : ∫ x in a..b, x ^ n = (b^(n+1) - a^(n+1)) / (n + 1) :=
+begin
+  have hderiv : deriv (λ x : ℝ, x^(n + 1) / (n + 1)) = λ x, x ^ n,
+  { have hne : (n + 1 : ℝ) ≠ 0 := by exact_mod_cast nat.succ_ne_zero n,
+    ext,
+    simp [mul_div_assoc, mul_div_cancel' _ hne] },
+  rw integral_deriv_eq_sub' _ hderiv;
+  norm_num [div_sub_div_same, (continuous_pow n).continuous_on],
+end
+
+@[simp]
+lemma integral_id : ∫ x in a..b, x = (b^2 - a^2) / 2 :=
+by simpa using integral_pow 1
+
+@[simp]
+lemma integral_one : ∫ x in a..b, (1:ℝ) = b - a :=
+by simpa using integral_pow 0
+
+@[simp]
+lemma integral_exp : ∫ x in a..b, exp x = exp b - exp a :=
+by rw integral_deriv_eq_sub'; norm_num [continuous_exp.continuous_on]
+
+@[simp]
+lemma integral_inv (h : (0:ℝ) ∉ interval a b) : ∫ x in a..b, x⁻¹ = log (b / a) :=
+begin
+  have h' := λ x hx, ne_of_mem_of_not_mem hx h,
+  rw [integral_deriv_eq_sub' _ deriv_log' (λ x hx, differentiable_at_log (h' x hx))
+        (continuous_on_inv'.mono (subset_compl_singleton_iff.mpr h)),
+      log_div (h' b right_mem_interval) (h' a left_mem_interval)],
+end
+
+@[simp]
+lemma integral_inv_of_pos (ha : 0 < a) (hb : 0 < b) : ∫ x in a..b, x⁻¹ = log (b / a) :=
+integral_inv $ not_mem_interval_of_lt ha hb
+
+@[simp]
+lemma integral_inv_of_neg (ha : a < 0) (hb : b < 0) : ∫ x in a..b, x⁻¹ = log (b / a) :=
+integral_inv $ not_mem_interval_of_gt ha hb
+
+lemma integral_one_div (h : (0:ℝ) ∉ interval a b) : ∫ x : ℝ in a..b, 1/x = log (b / a) :=
+by simp only [one_div, integral_inv h]
+
+lemma integral_one_div_of_pos (ha : 0 < a) (hb : 0 < b) : ∫ x : ℝ in a..b, 1/x = log (b / a) :=
+by simp only [one_div, integral_inv_of_pos ha hb]
+
+lemma integral_one_div_of_neg (ha : a < 0) (hb : b < 0) : ∫ x : ℝ in a..b, 1/x = log (b / a) :=
+by simp only [one_div, integral_inv_of_neg ha hb]
+
+@[simp]
+lemma integral_sin : ∫ x in a..b, sin x = cos a - cos b :=
+by rw integral_deriv_eq_sub' (λ x, -cos x); norm_num [continuous_on_sin]
+
+@[simp]
+lemma integral_cos : ∫ x in a..b, cos x = sin b - sin a :=
+by rw integral_deriv_eq_sub'; norm_num [continuous_on_cos]
+
+@[simp]
+lemma integral_inv_one_add_sq : ∫ x : ℝ in a..b, (1 + x^2)⁻¹ = arctan b - arctan a :=
+begin
+  simp only [← one_div],
+  refine integral_deriv_eq_sub' _ _ _ (continuous_const.div _ (λ x, _)).continuous_on;
+  norm_num,
+  { continuity },
+  { nlinarith },
+end
+
+lemma integral_one_div_one_add_sq : ∫ x : ℝ in a..b, 1 / (1 + x^2) = arctan b - arctan a :=
+by simp
+
+
+-- Simple integrals are now computable by `norm_num`. Here are some examples:
```

Comment:
There have been many discussions in the past about whether we should allow `example` in `src`. I really don't have time to reopen these arguments, but I'll point out that there are [files in `/archive`](https://github.com/leanprover-community/mathlib/blob/master/archive/examples/mersenne_primes.lean) with similar content :smile:

Reply:
We also have a bunch of derivative examples in [`test/differentiable.lean`](https://github.com/leanprover-community/mathlib/blob/master/test/differentiable.lean) so something in `test/` would be an option too.

Reply:
I like the idea of putting them in `test` (let's say, in a new file in that directory), with a pointer there in the module docstring.

Reply:
I hadn't known that those were options; by all means, they should go wherever is most appropriate.

Reply:
`test` sounds great to me!

Reply:
Okay, so I will remove the examples from this PR and add the file `test/integration.lean` in a future PR (it will be easiest for me to do it this way since these examples will anyway be subsumed by examples made possible by future PRs, e.g. #6357).